neo/Ledger/Blockchain.cs

```diff
@@ -405,6 +406,19 @@ private void Persist(Block block)
             {
                 List<ApplicationExecuted> all_application_executed = new List<ApplicationExecuted>();
                 snapshot.PersistingBlock = block;
+                using (ApplicationEngine engine = new ApplicationEngine(TriggerType.System, null, snapshot, 0, true))
+                {
+                    using (ScriptBuilder sb = new ScriptBuilder())
+                    {
+                        sb.EmitAppCall(NativeContract.GAS.ScriptHash, "distributeFees");
+                        engine.LoadScript(sb.ToArray());
+                    }
+                    engine.Execute();
+                    if (engine.State != VMState.HALT) throw new InvalidOperationException();
```

Comment:
How would a native function fail?

Reply:
This should not fail. If it fails, it means that there is an unrecoverable exception and should stop immediately.